src/particles/beam/BeamParticleContainerInit.cpp

```diff
@@ -60,190 +60,243 @@ namespace
 
 void
 BeamParticleContainer::
-InitBeamFixedPPC (const amrex::IntVect& a_num_particles_per_cell,
-                  const GetInitialDensity& get_density,
-                  const GetInitialMomentum& get_momentum,
-                  const amrex::Geometry& a_geom,
-                  const amrex::Real a_zmin,
-                  const amrex::Real a_zmax,
-                  const amrex::Real a_radius,
-                  const amrex::RealVect a_position_mean,
-                  const amrex::Real a_min_density,
-                  const amrex::Vector<int>& random_ppc)
+InitBeamFixedPPC3D ()
 {
-    HIPACE_PROFILE("BeamParticleContainer::InitParticles()");
+    HIPACE_PROFILE("BeamParticleContainer::InitBeamFixedPPC3D()");
 
     if (!Hipace::HeadRank()) { return; }
 
-    const amrex::IntVect ncells = a_geom.Domain().length();
-    amrex::Long ncells_total = (amrex::Long) ncells[0] * ncells[1] * ncells[2];
-    if ( ncells_total / Hipace::m_beam_injection_cr / Hipace::m_beam_injection_cr
-         > std::numeric_limits<int>::max() / 100 ){
-        amrex::Print()<<"WARNING: the number of cells is close to overflowing the maximum int,\n";
-        amrex::Print()<<"consider using a larger hipace.beam_injection_cr\n";
```

Comment:
With the per slice init, the `m_beam_injection_cr` is deprecated. Could you please fully remove it from the code and add it to the deprecated input paramters?

Reply:
I fully removed it